File: python/strings/FindString.scala
import java.util.Scanner
import scala.annotation.tailrec

object FindString {

  def findString2(s: String, sub: String): Int = {
    if(s.length >= sub.length) {
      if(s.substring(0, sub.length) == sub)
        1 + findString(s.drop(1), sub)
      else
        findString(s.drop(1), sub)
    } else {
      0
    }
  }

  def findString(string: String, sub: String): Int = {
    @tailrec
    def aux(s: String, acc: Int): Int = {
      if(s.length >= sub.length) {
        if(s.substring(0, sub.length) == sub)
          aux(s.drop(1), acc+1)
        else
          aux(s.drop(1), acc)
      } else {
        acc
      }
    }
    aux(string, 0)
  }

  def main(args: Array[String]) {
    val sc = new Scanner(System.in)
    val s = sc.nextLine()
    val sub = sc.nextLine()
    println(findString(s, sub))
  }
}

File: python/strings/find_string.ml
module S = String

let rec find_string s r =
  if (S.length s) >= (S.length r) then
    let next = S.sub s 1 ((S.length s)-1) in
    if (S.sub s 0 (S.length r)) = r then
      1 + find_string next r
    else
      find_string next r
  else
    0

let find_string string r =
  let rec aux s acc =
    if (S.length s) >= (S.length r) then
      let next = S.sub s 1 ((S.length s)-1) in
      if (S.sub s 0 (S.length r)) = r then
        aux next (acc+1)
      else
        aux next acc
    else acc
  in
  aux string 0

let () =
  let f x y = (x, y) in
  let (s, r) = Scanf.bscanf Scanf.Scanning.stdin "%s %s" f in
  Printf.printf "%i\n" (find_string s r)
